x-pack/plugins/ml/public/explorer/explorer_charts/components/explorer_chart_label/explorer_chart_label.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import './styles/explorer_chart_label.less';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {
+  EuiIconTip
+} from '@elastic/eui';
+
+import { ExplorerChartLabelBadge } from './explorer_chart_label_badge';
+import { ExplorerChartTooltip } from '../../explorer_chart_tooltip';
+
+export function ExplorerChartLabel({ detectorLabel, entityFields, infoTooltip, wrapLabel }) {
+  /*
+    Depending on whether we wrap the entityField badges to a new line, we render this differently:
+
+    1. All in one line:
+        <detectorLabel> - <entityBadge1> <entityBadge2> ... <infoIcon>
+
+    2. Multiple lines:
+        <detectorLabel> <infoIcon> <br />
+        <entityBadge1> <entityBadge2> ...
+  */
+
+  const labelSeparator = (wrapLabel === true) ? <br /> : ' - ';
+
+  const entityFieldBadges = entityFields.map((entity, j) => {
+    return <span key={j}><ExplorerChartLabelBadge entity={entity} /> </span>;
+  });
+
```

Comment:
Just a quick note - we generally want to avoid using indexes as keys and instead use a unique identifier.

Since React uses the key to identify DOM elements and see if they've changed. So if the order changes for some reason there could be bugs due to React thinking that because the key is the same, the content is the same. (more info here if you're curious: https://medium.com/@vraa/why-using-an-index-as-key-in-react-is-probably-a-bad-idea-7543de68b17c)

I haven't dug as deeply for fragments regarding this but I'd err on the side of caution.

Reply:
Nice find, I built a custom key for the outer `.map()` part in the same file but let hat slip through here :)

Reply:
@alvarezmelissa87 pushed an update to use a proper key based on the `entity` data.